src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectAuthenticationHandler.cs

```diff
@@ -217,6 +218,47 @@ public OpenIdConnectAuthenticationHandler(HttpClient backchannel)
 
             message.State = Options.StateDataFormat.Protect(properties);
 
+            if (Options.UsePostAuthorizationRequests)
+            {
+                using (var buffer = new MemoryStream())
+                using (var writer = new StreamWriter(buffer))
+                {
+                    writer.WriteLine("<!doctype html>");
+                    writer.WriteLine("<html>");
+                    writer.WriteLine("<body>");
+
+                    writer.WriteLine("<form name='form' method='post' action='" + Options.HtmlEncoder.HtmlEncode(message.IssuerAddress) + "'>");
```

Comment:
Note: `OpenIdConnectMessage`, which derives from `AuthenticationProtocolMessage`, has a `BuildFormPost` method that is supposed to do exactly the same thing. Sadly it's totally busted as it internally uses `Uri.EscapeDataString` which messes up the action attribute. Instead, I used the "manual" approach I already used for `AspNet.Security.OpenIdConnect.Server`: https://github.com/aspnet-contrib/AspNet.Security.OpenIdConnect.Server/blob/vNext/src/AspNet.Security.OpenIdConnect.Server/OpenIdConnectServerHandler.cs#L709